contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@scroll-tech/contracts/L1/gateways/IL1GatewayRouter.sol";
+import "@scroll-tech/contracts/L1/IL1ScrollMessenger.sol";
+import "./interfaces/AdapterInterface.sol";
+
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+    uint32 public immutable l2GasLimit = 250_000;
+
+    IL1GatewayRouter public immutable l1GatewayRouter;
+    IL1ScrollMessenger public immutable l1ScrollMessenger;
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1GatewayRouter Standard bridge contract.
+     * @param _l1ScrollMessenger Scroll Messenger contract.
+     */
+    constructor(IL1GatewayRouter _l1GatewayRouter, IL1ScrollMessenger _l1ScrollMessenger) {
+        l1GatewayRouter = _l1GatewayRouter;
+        l1ScrollMessenger = _l1ScrollMessenger;
+    }
+
+    /**
+     * @notice Send message to `target` on L2.
+     * @dev This method is marked payable because relaying the message might require a fee
+     * to be paid by the sender to forward the message to L2. However, it will not send msg.value
+     * to the target contract on L2.
+     * @param target L2 address to send message to.
+     * @param message Message to send to `target`.
+     */
+    function relayMessage(address target, bytes calldata message) external payable {
+        // We can specifically send a message with 0 value to the Scroll Bridge
+        // and it will not forward any ETH to the target contract on L2. However,
+        // we need to set the payable value to msg.value to ensure that the Scroll
+        // Bridge has enough gas to forward the message to L2.
```

Comment:
Are we able to compute the cost of l2Gas? For instance, if we know we're sending 250k L2 gas, are we able to compute the cost of that gas in the transaction so we don't get a refund back (or underpay)?

Reply:
I can't seem to find any explicit docs - I will forward via TG

Reply:
If they are giving a synchronous refund, there should be some way to compute the cost -- hopefully just a public function.

Reply:
Done - we resolved this offline @mrice32 but now we are referencing the l2gas oracle on mainnet to get the max fee to send to the contracts.